lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -65,3 +69,88 @@ function dosomething_kudos_get_kudos_query($params = [], $count = NULL, $start =
   $result = $query->execute();
   return array_keys($result['kudos']);
 }
+
+
+/**
+ * Sort kudos collection based on taxonomy term and then
+ * collect and group them by term.
+ *
+ * @param array $data
+ *
+ * @return array
+ */
+function dosomething_kudos_sort($data) {
+  usort($data, 'dosomething_kudos_sort_by_taxonomy_term');
```

Comment:
@weerd Sorry if I'm missing something obvious, but I don't get why we're sorting by taxonomy term here.

Doesn't the grouping by term happen via the new line you added to `dosomething_reportback_get_reportback_files_query`:
`$query->addExpression('GROUP_CONCAT(DISTINCT k.kid)', 'kids');` ?